### DISCUSSION TITLE
external css and js add.

### DISCUSSION THREAD

**Discussion by akashgh72** [adonisjs/core#3143] [Help] [answered]
I have added css and js file under public assets folder but after running "node ace serve --watch" all files are getting deleted. How can i integrate a admin panel css and js into it. You can check it from below screenshot.

![Screenshot 2021-09-07 at 5 34 46 PM](https://user-images.githubusercontent.com/90257417/132341568-3708cc35-dec0-468d-ac12-b8b3ce671f67.png)

**Comment by thetutlage** [comment, marked as answer]
You don't keep files under the public directory. The public directory is the output target when using Webpack Encore. Keep them inside the `resources` directory

**Reply by etc-austin** [reply]
How are you supposed to call a css file? I put style.css in the resource/css directory and used the helper
`<link rel="stylesheet" type="text/css" href="{{ asset('assets/style.css') }}"> 
`
I get 500 error that style cannot be found

**Reply by RomainLanz** [reply]
Use `Encore.copyFiles()` to copies your file to the `public` folder.

**Reply by muasx88** [reply]
hai, any example of using encore to copy css external file to public?

**Reply by CrowBe** [reply]
@muasx88 Some info here: https://docs.adonisjs.com/guides/assets-manager
Out of the box you can use the adonis `node ace configure encore` once you have encore installed.
Generally speaking, you will be using the one of the css loaders found inside the webpack config file generated by the above command.
You can also see this documentation for more information on copying files with Encore, if that isn't sufficient.
https://symfony.com/doc/current/frontend/encore/copy-files.html#referencing-image-files-from-a-template
